File: 7KYU/Strong Number (Special Numbers Series #2)/solution.clj
(ns kata)

(defn factorial
  ([n] (factorial n 1))
  ([n sum]
   (cond (= 0 n) sum
         :else (recur (dec n) (* sum n)))))

(defn strong [n]
  (if (= n
         (->> n str
              (map (comp read-string str))
              (map factorial)
              (reduce +)))
    "STRONG!!!!"
    "Not Strong !!"))

File: 7KYU/Strong Number (Special Numbers Series #2)/solution.ts
export const factorial = (n: number): number => [...Array(n+1).keys()].slice(1).reduce((acc: number, i: number) => acc * i, 1);

export const strongNumber = (n: number): string => {
  var f = String(n).split('')
                   .map(Number)
                   .reduce((acc: number, i: number) => acc + factorial(i), 0);
  return f === n ? "STRONG!!!!" : "Not Strong !!" ;
};